.github/workflows/ci-gold-test.yml

```diff
@@ -0,0 +1,23 @@
+name: gold test
```

Comment:
This is a bit of a general question about our testing philosophy here. I find it confusing that not all tests are in the /tests directory, but maybe that's fine. Maybe we should call that directory e2e-test or golden-tests? Or some other name that makes it clear this is only one kind of tests, not all tests.

Reply:
Fair point. I think I'd like to stick with the Google convention to have unit tests live right next to the code it tests, so I am renaming the test directory to e2e-test

Reply:
Thanks